lib/node_modules/@stdlib/_tools/eslint/rules/doctest-annotation-spacing/lib/main.js

```diff
@@ -0,0 +1,194 @@
+var rule;
+
+
+// FUNCTIONS //
+
+/**
+* Rule for validating that return annotations have correct spacing.
+*
+* @param {Object} context - ESLint context
+* @returns {Object} validators
+*/
+function main( context ) {
+	var options;
+	var source;
+	var opts;
+
+	opts = copy( DEFAULTS );
+	options = context.options[ 0 ];
+	if ( options ) {
+		if ( hasOwnProp( options, 'numSpaces' ) ) {
+			opts.numSpaces = options.numSpaces;
+		}
+	}
+	source = context.getSourceCode();
+
+	/**
+	* Reports the error message.
+	*
+	* @private
+	* @param {Object} loc - error location info
+	* @param {string} msg - error message
+	*/
+	function report( loc, msg ) {
+		context.report({
+			'node': null,
+			'message': msg,
+			'loc': loc
+		});
+	}
+
+	/**
+	* Checks whether return annotations in the current program have correct spacing.
+	*
+	* @private
+	* @param {ASTNode} node - node to examine
+	*/
+	function validate( node ) {
+		var comments;
+		var i;
+
+		comments = source.getAllComments( node );
+		for ( i = 0; i < comments.length; i++ ) {
+			checkComment( comments[ i ] );
+		}
+	}
+
+	/**
+	* Checks whether a comment is a return annotation and, if so, whether it has correct spacing.
+	*
+	* @private
+	* @param {Object} comment - comment to examine
+	*/
+	function checkComment( comment ) {
+		var spacingBefore;
+		var spacingAfter;
+		var expected;
+		var matches;
+		var keyword;
+		var prefix;
+		var text;
+		var msg;
+
+		// Only check single-line comments:
+		if ( comment.type !== 'Line' ) {
+			return;
+		}
+		// Reconstruct the comment text with the leading slashes:
+		text = '//' + comment.value;
+
+		matches = text.match( RE_ANNOTATION );
+		if ( matches ) {
+			spacingBefore = matches[ 1 ].length;
+			prefix = matches[ 2 ] || '';  // 'e.g.,' or empty
+			keyword = matches[ 3 ];       // 'returns' or '=>'
+			spacingAfter = matches[ 4 ].length;
+
+			// Check spacing before keyword (must be exactly 1 space):
+			if ( spacingBefore !== 1 ) {
+				if ( prefix ) {
+					msg = 'Return annotation `' + prefix + ' ' + keyword + '` should have exactly 1 space after `//`. Found ' + spacingBefore + ' space(s).';
+				} else {
+					msg = 'Return annotation `' + keyword + '` should have exactly 1 space after `//`. Found ' + spacingBefore + ' space(s).';
+				}
+				report( comment.loc, msg );
+			}
+			// Check spacing after keyword (configurable):
+			expected = opts.numSpaces;
+			if ( spacingAfter !== expected ) {
+				msg = 'Return annotation `' + keyword + '` should be followed by ' + expected + ' space(s). Found ' + spacingAfter + ' space(s).';
+				report( comment.loc, msg );
+			}
+		}
+	}
+
+	return {
+		'Program': validate
+	};
+}
+
+
+// MAIN //
+
+rule = {
+	'meta': {
+		'docs': {
+			'description': 'enforce spacing after return annotations in single-line comments'
+		},
+		'schema': [
+			{
+				'type': 'object',
+				'properties': {
+					'numSpaces': {
+						'type': 'integer',
+						'minimum': 1
```

Comment:
If we want this to be a "general" thing eventually which others can use, I'd allow the minimum to be `0`.